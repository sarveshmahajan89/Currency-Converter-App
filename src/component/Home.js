import React from "react";
import Conversion from './Conversion';

class Home  extends React.Component {
    render() {
        return (
            <React.Fragment>
                <Conversion />
            </React.Fragment>
        );
    }
}
export default Home ;