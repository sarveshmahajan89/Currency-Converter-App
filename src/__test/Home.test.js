import React from 'react';

import Home from '../component/Home.js'
import {shallow} from "enzyme/build/index";

describe('Home', function() {
    it('Home renders correctly', () => {

        expect(<Home/>).toMatchSnapshot();
    });

    it('should render Home correctly with shallow', function () {

        const component = shallow(<Home/>);
        expect(component).toMatchSnapshot();
    });

    it('should render Home correctly Conversion', function () {

        const component = shallow(<Home/>);
        expect(component.exists('Conversion')).toBe(true);
    });
});