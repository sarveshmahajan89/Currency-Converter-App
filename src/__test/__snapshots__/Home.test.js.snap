// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home renders correctly 1`] = `<Home />`;

exports[`Home should render Home correctly with shallow 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <React.Fragment>
    <Conversion />
  </React.Fragment>,
  "nodes": Array [
    <React.Fragment>
      <Conversion />
    </React.Fragment>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_didScheduleRenderPhaseUpdate": false,
    "_dispatcher": Object {
      "readContext": [Function],
      "useCallback": [Function],
      "useContext": [Function],
      "useDebugValue": [Function],
      "useDeferredValue": [Function],
      "useEffect": [Function],
      "useImperativeHandle": [Function],
      "useLayoutEffect": [Function],
      "useMemo": [Function],
      "useReducer": [Function],
      "useRef": [Function],
      "useResponder": [Function],
      "useState": [Function],
      "useTransition": [Function],
    },
    "_element": <Home />,
    "_firstWorkInProgressHook": null,
    "_forcedUpdate": false,
    "_instance": Home {
      "context": Object {},
      "props": Object {},
      "refs": Object {},
      "state": null,
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_isReRender": false,
    "_newState": null,
    "_numberOfReRenders": 0,
    "_renderPhaseUpdates": null,
    "_rendered": <React.Fragment>
      <Conversion />
    </React.Fragment>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "_workInProgressHook": null,
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Home />,
}
`;
